### PR TITLE
Test/images api tests - add test suite for Images API endpoints

### LINKED ISSUE #853: TEST: Add missing test suite for Images API endpoints
## Is there an existing issue for this?
- [x] I have searched the existing issues

## Description

The `backend/tests/` directory contains comprehensive test files for:
- `test_folders.py` - Folders API tests
- `test_albums.py` - Albums API tests  
- `test_face_clusters.py` - Face Clusters API tests
- `test_user_preferences.py` - User Preferences API tests
However, there is **no test file for the Images API** (`backend/app/routes/images.py`).

## Current State
The Images API has the following endpoints that are untested:
- `GET /images/` - Get all images with optional tagged filter
- `POST /images/toggle-favourite` - Toggle favourite status

## Proposed Solution

Create `backend/tests/test_images.py` with comprehensive test coverage:
### Test Cases to Cover

**GET /images/ endpoint:**
- [ ] Success with multiple images
- [ ] Empty database handling
- [ ] Filter by tagged status (true/false)
- [ ] Database error handling

**POST /images/toggle-favourite endpoint:**
- [ ] Successful toggle
- [ ] Image not found handling
- [ ] Missing image_id validation
- [ ] Database error handling

**Edge Cases:**
- [ ] Null/empty metadata handling
- [ ] GPS location metadata
- [ ] Integration workflow tests

## Benefits

- Improves test coverage for the backend
- Catches regressions in image-related functionality
- Follows the existing testing patterns in the project
- Helps maintain code quality during future development

## Checklist
- [x] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by Rudra-Tiwari-codes**
## Summary
Fixes #853 
Creates a comprehensive test suite for the Images API endpoints, filling a gap in the backend test coverage.
## Changes Made
**New File:** `backend/tests/test_images.py`

### Test Coverage
| Category | Test Cases |
|----------|------------|
| GET /images/ | 5 tests (success, empty, filtered, errors) |
| POST /toggle-favourite | 4 tests (success, not found, validation, errors) |
| Edge Cases | 3 tests (null metadata, GPS data, workflows) |
| **Total** | **15+ test cases** |

### Test Structure

```python
class TestImagesAPI:
    # GET /images/ tests
    def test_get_all_images_success(...)
    def test_get_all_images_empty(...)
    def test_get_all_images_filter_tagged(...)
    def test_get_all_images_filter_untagged(...)
    def test_get_all_images_database_error(...)
    
    # POST /toggle-favourite tests
    def test_toggle_favourite_success(...)
    def test_toggle_favourite_not_found(...)
    def test_toggle_favourite_missing_image_id(...)
    def test_toggle_favourite_database_error(...)
    
    # Edge cases
    def test_get_images_with_null_metadata(...)
    def test_get_images_with_location_metadata(...)
    def test_toggle_and_verify_favourite(...)

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

* **Documentation**
  * Enhanced internal documentation for image detection utilities.

* **Tests**
  * Added comprehensive test coverage for Images API endpoints, including retrieval, favorite toggling, and edge case handling.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

### CODE REVIEW

**Comment by fransafu** [comment]
**Warning:** This author is forking multiple ML projects such as google-deepmind/alphafold, ml-explore/mlx, openai/CLIP, pytorch/pytorch, tensorflow/tensorflow, anthropics/claude-code, vllm-project/vllm, and others, adding minimal "contributions" (often for tests or miscellaneous changes) without proper validation. A review of their commits shows mostly local implementations of TODOs copied from existing projects, with little to no substantive review or testing.

So far, this author has forked **41 repositories** following the same pattern. Be careful when accepting this PR. It’s also concerning how this author is able to submit PRs across four repositories in the same day, each requiring large context, which strongly suggests a highly automated workflow.

**Comment by Rudra-Tiwari-codes** [comment]
Thank you for the feedback regarding my recent activity. I am a student and I have been using these smaller tasks as a way to familiarize myself with the architecture of various codebases. I was not aware that submitting multiple minor pull requests was considered disruptive to the maintainer workflow or seen as contribution padding. I appreciate the correction and will pay much closer attention to the impact of my work moving forward. I am closing this pull request now to focus on delivering more substantive technical contributions that provide genuine value to the community.
